sdk/go/pulumi/alias.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pulumi
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Alias is a partial description of prior named used for a resource. It can be processed in the
+// context of a resource creation to determine what the full aliased URN would be.
+type Alias struct {
+	// The URN that uniquely identifies a resource
```

Comment:
Nit: state that this is optional + how it conflicts (or doesn't) with other options